libcloud/test/compute/test_gce.py

```diff
@@ -214,6 +214,15 @@ def test_build_service_account_gce_struct(self):
         self.assertTrue('email' in actual)
         self.assertTrue('scopes' in actual)
 
+    def test_build_service_account_with_email_gce_struct(self):
+        self.assertRaises(ValueError,
+                          self.driver._build_service_account_gce_struct, None)
+        input = {'scopes': ['compute-ro'], 'email': 'test@test.com' }
+        actual = self.driver._build_service_account_with_email_gce_struct(input)
+        self.assertTrue('email' in actual)
+        self.assertEqual(actual['email'], 'test@test.com')
+        self.assertTrue('scopes' in actual)
```

Comment:
I think after adding the `else` part you should check both 'default' and user provided email. Something like,
```
input = {'scopes': ['compute-ro']}
actual = self.driver._build_service_account_with_email_gce_struct(input) 
self.assertTrue('email' in actual)
self.assertEqual(actual['email'], 'default')
self.assertTrue('scopes' in actual)

input = {'scopes': ['compute-ro'], 'email': 'test@test.com'}
actual = self.driver._build_service_account_with_email_gce_struct(input) 
self.assertTrue('email' in actual)
self.assertEqual(actual['email'], 'test@test.com')
self.assertTrue('scopes' in actual)
```

Reply:
Are your tests passing locally?